File: plugin/src/main/java/com/stratio/cassandra/lucene/schema/analysis/StandardAnalyzers.java
/*
 * Licensed to STRATIO (C) under one or more contributor license agreements.
 * See the NOTICE file distributed with this work for additional information
 * regarding copyright ownership.  The STRATIO (C) licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package com.stratio.cassandra.lucene.schema.analysis;

import org.apache.lucene.analysis.Analyzer;
import org.apache.lucene.analysis.ar.ArabicAnalyzer;
import org.apache.lucene.analysis.bg.BulgarianAnalyzer;
import org.apache.lucene.analysis.br.BrazilianAnalyzer;
import org.apache.lucene.analysis.ca.CatalanAnalyzer;
import org.apache.lucene.analysis.cjk.CJKAnalyzer;
import org.apache.lucene.analysis.ckb.SoraniAnalyzer;
import org.apache.lucene.analysis.core.KeywordAnalyzer;
import org.apache.lucene.analysis.core.SimpleAnalyzer;
import org.apache.lucene.analysis.core.StopAnalyzer;
import org.apache.lucene.analysis.core.WhitespaceAnalyzer;
import org.apache.lucene.analysis.cz.CzechAnalyzer;
import org.apache.lucene.analysis.da.DanishAnalyzer;
import org.apache.lucene.analysis.de.GermanAnalyzer;
import org.apache.lucene.analysis.el.GreekAnalyzer;
import org.apache.lucene.analysis.en.EnglishAnalyzer;
import org.apache.lucene.analysis.es.SpanishAnalyzer;
import org.apache.lucene.analysis.eu.BasqueAnalyzer;
import org.apache.lucene.analysis.fa.PersianAnalyzer;
import org.apache.lucene.analysis.fi.FinnishAnalyzer;
import org.apache.lucene.analysis.fr.FrenchAnalyzer;
import org.apache.lucene.analysis.ga.IrishAnalyzer;
import org.apache.lucene.analysis.gl.GalicianAnalyzer;
import org.apache.lucene.analysis.hi.HindiAnalyzer;
import org.apache.lucene.analysis.hu.HungarianAnalyzer;
import org.apache.lucene.analysis.hy.ArmenianAnalyzer;
import org.apache.lucene.analysis.id.IndonesianAnalyzer;
import org.apache.lucene.analysis.it.ItalianAnalyzer;
import org.apache.lucene.analysis.lv.LatvianAnalyzer;
import org.apache.lucene.analysis.nl.DutchAnalyzer;
import org.apache.lucene.analysis.no.NorwegianAnalyzer;
import org.apache.lucene.analysis.pt.PortugueseAnalyzer;
import org.apache.lucene.analysis.ro.RomanianAnalyzer;
import org.apache.lucene.analysis.ru.RussianAnalyzer;
import org.apache.lucene.analysis.standard.ClassicAnalyzer;
import org.apache.lucene.analysis.standard.StandardAnalyzer;
import org.apache.lucene.analysis.sv.SwedishAnalyzer;
import org.apache.lucene.analysis.th.ThaiAnalyzer;
import org.apache.lucene.analysis.tr.TurkishAnalyzer;

import java.util.Locale;

/**
 * Prebuilt Lucene {@link Analyzer}s that can be instantiated by name.
 */
public enum StandardAnalyzers {

    STANDARD() {
        @Override
        protected Analyzer build() {
            return new StandardAnalyzer();
        }
    },

    DEFAULT() {
        @Override
        protected Analyzer build() {
            return STANDARD.build();
        }
    },

    KEYWORD() {
        @Override
        protected Analyzer build() {
            return new KeywordAnalyzer();
        }
    },

    STOP {
        @Override
        protected Analyzer build() {
            return new StopAnalyzer();

        }
    },

    WHITESPACE {
        @Override
        protected Analyzer build() {
            return new WhitespaceAnalyzer();

        }
    },

    SIMPLE {
        @Override
        protected Analyzer build() {
            return new SimpleAnalyzer();

        }
    },

    CLASSIC {
        @Override
        protected Analyzer build() {
            return new ClassicAnalyzer();

        }
    },

    ARABIC {
        @Override
        protected Analyzer build() {
            return new ArabicAnalyzer();
        }
    },

    ARMENIAN {
        @Override
        protected Analyzer build() {
            return new ArmenianAnalyzer();
        }
    },

    BASQUE {
        @Override
        protected Analyzer build() {
            return new BasqueAnalyzer();
        }
    },

    BRAZILIAN {
        @Override
        protected Analyzer build() {
            return new BrazilianAnalyzer();
        }
    },

    BULGARIAN {
        @Override
        protected Analyzer build() {
            return new BulgarianAnalyzer();
        }
    },

    CATALAN {
        @Override
        protected Analyzer build() {
            return new CatalanAnalyzer();
        }
    },

    CHINESE() {
        @Override
        protected Analyzer build() {
            return new StandardAnalyzer();
        }
    },

    CJK {
        @Override
        protected Analyzer build() {
            return new CJKAnalyzer();
        }
    },

    CZECH {
        @Override
        protected Analyzer build() {
            return new CzechAnalyzer();
        }
    },

    DUTCH {
        @Override
        protected Analyzer build() {
            return new DutchAnalyzer();

        }
    },

    DANISH {
        @Override
        protected Analyzer build() {
            return new DanishAnalyzer();

        }
    },

    ENGLISH {
        @Override
        protected Analyzer build() {
            return new EnglishAnalyzer();
        }
    },

    FINNISH {
        @Override
        protected Analyzer build() {
            return new FinnishAnalyzer();
        }
    },

    FRENCH {
        @Override
        protected Analyzer build() {
            return new FrenchAnalyzer();
        }
    },

    GALICIAN {
        @Override
        protected Analyzer build() {
            return new GalicianAnalyzer();
        }
    },

    GERMAN {
        @Override
        protected Analyzer build() {
            return new GermanAnalyzer();
        }
    },

    GREEK {
        @Override
        protected Analyzer build() {
            return new GreekAnalyzer();
        }
    },

    HINDI {
        @Override
        protected Analyzer build() {
            return new HindiAnalyzer();

        }
    },

    HUNGARIAN {
        @Override
        protected Analyzer build() {
            return new HungarianAnalyzer();
        }
    },

    INDONESIAN {
        @Override
        protected Analyzer build() {
            return new IndonesianAnalyzer();
        }
    },

    IRISH {
        @Override
        protected Analyzer build() {
            return new IrishAnalyzer();
        }
    },

    ITALIAN {
        @Override
        protected Analyzer build() {
            return new ItalianAnalyzer();
        }
    },

    LATVIAN {
        @Override
        protected Analyzer build() {
            return new LatvianAnalyzer();
        }
    },

    NORWEGIAN {
        @Override
        protected Analyzer build() {
            return new NorwegianAnalyzer();
        }
    },

    PERSIAN {
        @Override
        protected Analyzer build() {
            return new PersianAnalyzer();
        }
    },

    PORTUGUESE {
        @Override
        protected Analyzer build() {
            return new PortugueseAnalyzer();
        }
    },

    ROMANIAN {
        @Override
        protected Analyzer build() {
            return new RomanianAnalyzer();

        }
    },

    RUSSIAN {
        @Override
        protected Analyzer build() {
            return new RussianAnalyzer();
        }
    },

    SORANI {
        @Override
        protected Analyzer build() {
            return new SoraniAnalyzer();
        }
    },

    SPANISH {
        @Override
        protected Analyzer build() {
            return new SpanishAnalyzer();
        }
    },

    SWEDISH {
        @Override
        protected Analyzer build() {
            return new SwedishAnalyzer();
        }
    },

    TURKISH {
        @Override
        protected Analyzer build() {
            return new TurkishAnalyzer();
        }
    },

    THAI {
        @Override
        protected Analyzer build() {
            return new ThaiAnalyzer();
        }
    };

    /**
     * Returns a new instance of the defined {@link Analyzer}.
     *
     * @return a new analyzer
     */
    abstract Analyzer build();

    /**
     * Returns the prebuilt {@link Analyzer} identified by the specified name, or {@code null} if there is no such
     * {@link Analyzer}.
     *
     * @param name a prebuilt {@link Analyzer} name
     * @return the analyzer, or {@code null} if there is no such analyzer
     */
    public static Analyzer get(String name) {
        try {
            return valueOf(name.toUpperCase(Locale.ROOT)).get();
        } catch (IllegalArgumentException e) {
            return null;
        }
    }

    /**
     * Returns the {@link Analyzer} defined by this.
     *
     * @return the analyzer
     */
    public Analyzer get() {
        return build();
    }

}

File: plugin/src/main/scala/com/stratio/cassandra/lucene/schema/analysis/StandardAnalyzers.scala
/*
 * Copyright (C) 2014 Stratio (http://stratio.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.stratio.cassandra.lucene.schema.analysis

import org.apache.lucene.analysis.Analyzer
import org.apache.lucene.analysis.ar.ArabicAnalyzer
import org.apache.lucene.analysis.bg.BulgarianAnalyzer
import org.apache.lucene.analysis.br.BrazilianAnalyzer
import org.apache.lucene.analysis.ca.CatalanAnalyzer
import org.apache.lucene.analysis.cjk.CJKAnalyzer
import org.apache.lucene.analysis.ckb.SoraniAnalyzer
import org.apache.lucene.analysis.core.{KeywordAnalyzer, SimpleAnalyzer, StopAnalyzer, WhitespaceAnalyzer}
import org.apache.lucene.analysis.cz.CzechAnalyzer
import org.apache.lucene.analysis.da.DanishAnalyzer
import org.apache.lucene.analysis.de.GermanAnalyzer
import org.apache.lucene.analysis.el.GreekAnalyzer
import org.apache.lucene.analysis.en.EnglishAnalyzer
import org.apache.lucene.analysis.es.SpanishAnalyzer
import org.apache.lucene.analysis.eu.BasqueAnalyzer
import org.apache.lucene.analysis.fa.PersianAnalyzer
import org.apache.lucene.analysis.fi.FinnishAnalyzer
import org.apache.lucene.analysis.fr.FrenchAnalyzer
import org.apache.lucene.analysis.ga.IrishAnalyzer
import org.apache.lucene.analysis.gl.GalicianAnalyzer
import org.apache.lucene.analysis.hi.HindiAnalyzer
import org.apache.lucene.analysis.hu.HungarianAnalyzer
import org.apache.lucene.analysis.hy.ArmenianAnalyzer
import org.apache.lucene.analysis.id.IndonesianAnalyzer
import org.apache.lucene.analysis.it.ItalianAnalyzer
import org.apache.lucene.analysis.lv.LatvianAnalyzer
import org.apache.lucene.analysis.nl.DutchAnalyzer
import org.apache.lucene.analysis.no.NorwegianAnalyzer
import org.apache.lucene.analysis.pt.PortugueseAnalyzer
import org.apache.lucene.analysis.ro.RomanianAnalyzer
import org.apache.lucene.analysis.ru.RussianAnalyzer
import org.apache.lucene.analysis.standard.{ClassicAnalyzer, StandardAnalyzer}
import org.apache.lucene.analysis.sv.SwedishAnalyzer
import org.apache.lucene.analysis.th.ThaiAnalyzer
import org.apache.lucene.analysis.tr.TurkishAnalyzer

/**
 * Prebuilt Lucene [[Analyzer]]s that can be instantiated by name.
 */
object StandardAnalyzers extends Enumeration {
    type StandardAnalyzers = Value
    val ARABIC = Value("arabic")
    val ARMENIAN = Value("armenian")
    val BASQUE = Value("basque")
    val BRAZILIAN = Value("brazilian")
    val BULGARIAN = Value("bulgarian")
    val CATALAN = Value("catalan")
    val CHINESE = Value("chinese")
    val CJK = Value("cjk")
    val CLASSIC = Value("classic")
    val CZECH = Value("czech")
    val DANISH = Value("danish")
    val DEFAULT = Value("default")
    val DUTCH = Value("dutch")
    val ENGLISH = Value("english")
    val FINNISH = Value("finnish")
    val FRENCH = Value("french")
    val GALICIAN = Value("galician")
    val GERMAN = Value("german")
    val GREEK = Value("greek")
    val HINDI = Value("hindi")
    val HUNGARIAN = Value("hungarian")
    val INDONESIAN = Value("indonesian")
    val IRISH = Value("irish")
    val ITALIAN = Value("italian")
    val KEYWORD = Value("keyword")
    val LATVIAN = Value("latvian")
    val NORWEGIAN = Value("norwegian")
    val PERSIAN = Value("persian")
    val PORTUGUESE = Value("portuguese")
    val ROMANIAN = Value("romanian")
    val RUSSIAN = Value("russian")
    val SIMPLE = Value("simple")
    val SORANI = Value("sorani")
    val SPANISH = Value("spanish")
    val STANDARD = Value("standard")
    val STOP = Value("stop")
    val SWEDISH = Value("swedish")
    val THAI = Value("thai")
    val TURKISH = Value("turkish")
    val WHITESPACE = Value("whitespace")

    /**
      * Returns the prebuilt [[Analyzer]] identified by the specified name, or {{{null}}} if there is no such
      * [[Analyzer]].
      *
      * @param name a prebuilt [[Analyzer]] name
      * @return the analyzer, or {{{null}}} if there is no such analyzer
      */
    def get(name: String): Analyzer = {

        val standardAnalyzer = Value(name.toLowerCase)
        standardAnalyzer match {
            case ARABIC => new ArabicAnalyzer()
            case ARMENIAN => new ArmenianAnalyzer()
            case BASQUE => new BasqueAnalyzer()
            case BRAZILIAN => new BrazilianAnalyzer()
            case BULGARIAN => new BulgarianAnalyzer()
            case CATALAN => new CatalanAnalyzer()
            case CHINESE => new StandardAnalyzer()
            case CJK => new CJKAnalyzer()
            case CLASSIC => new ClassicAnalyzer()
            case CZECH => new CzechAnalyzer()
            case DANISH => new DanishAnalyzer()
            case DEFAULT => new StandardAnalyzer()
            case DUTCH => new DutchAnalyzer()
            case ENGLISH => new EnglishAnalyzer()
            case FINNISH => new FinnishAnalyzer()
            case FRENCH => new FrenchAnalyzer()
            case GALICIAN => new GalicianAnalyzer()
            case GERMAN => new GermanAnalyzer()
            case GREEK => new GreekAnalyzer()
            case HINDI => new HindiAnalyzer()
            case HUNGARIAN => new HungarianAnalyzer()
            case INDONESIAN => new IndonesianAnalyzer()
            case IRISH => new IrishAnalyzer()
            case ITALIAN => new ItalianAnalyzer()
            case KEYWORD => new KeywordAnalyzer()
            case LATVIAN => new LatvianAnalyzer()
            case NORWEGIAN => new NorwegianAnalyzer()
            case PERSIAN => new PersianAnalyzer()
            case PORTUGUESE => new PortugueseAnalyzer()
            case ROMANIAN => new RomanianAnalyzer()
            case RUSSIAN => new RussianAnalyzer()
            case SIMPLE => new SimpleAnalyzer()
            case SORANI => new SoraniAnalyzer()
            case SPANISH => new SpanishAnalyzer()
            case STANDARD => new StandardAnalyzer()
            case STOP => new StopAnalyzer()
            case SWEDISH => new SwedishAnalyzer()
            case THAI => new ThaiAnalyzer()
            case TURKISH => new TurkishAnalyzer()
            case WHITESPACE => new WhitespaceAnalyzer()
            case _ => null
        }
    }
}